### PR TITLE
Feedback Modification

### PR DESCRIPTION
**Pull request by openbots-ff**
- Breakpoint button for commands 
- Hotkeys for breakpoint/enable/disable commands 
- Hotkeys for project tool strip menu 
- Clarification for creating vs writing to text files 
- Clear search button for command search box 
- Support for Dictionaries in debug 
- Search bars for Variable and Element Selectors 
- Deleting enclosing commands such as "BeginLoop" will delete all commands up to the respective End command.  
- User will be prompted for a new file/folder name when adding one to a project.

### CODE REVIEW

File: OpenBots.Commands/OpenBots.Commands.List/CreateListCommand.cs [hunk 1]
```diff
@@ -66,7 +66,7 @@ public override void RunCommand(object sender)
             dynamic vNewList = null;
             string[] splitListItems = null;
 
-            if (!string.IsNullOrEmpty(v_ListItems.Trim()))
```
Comment (openbots-ch) [inline OpenBots.Commands/OpenBots.Commands.List/CreateListCommand.cs:69]: Was trimming empty strings causing issues for this check? I would assume it lets you trim the spaces out of a technically blank string for asserting empty.

Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.List/CreateListCommand.cs:69]: it turns out that if left empty, the default value is null. And you can't trim null.

File: OpenBots.Studio/Commands/Input Commands/UIAutomationCommand.cs [hunk 2]
```diff
@@ -630,7 +630,7 @@ public void ShowRecorder(object sender, EventArgs e)
         {
             //get command reference
             //create recorder
-            frmThickAppElementRecorder newElementRecorder = new frmThickAppElementRecorder();
+            frmAdvancedUIElementRecorder newElementRecorder = new frmAdvancedUIElementRecorder();
```
Comment (openbots-ch) [inline OpenBots.Studio/Commands/Input Commands/UIAutomationCommand.cs:633]: Will this still work cleanly for single element grabbing, or will it go for a sequence every time?

**Review by openbots-ch — CHANGES_REQUESTED** [review]

File: OpenBots.Studio/Commands/Input Commands/UIAutomationCommand.cs (hunk 2 above)
Comment (openbots-ff) [reply OpenBots.Studio/Commands/Input Commands/UIAutomationCommand.cs:633]: Yes, there's a flag that sets it

File: OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs [hunk 3]
```diff
@@ -89,6 +89,10 @@ public class GetOutlookEmailsCommand : ScriptCommand
         [Remarks("Variables not pre-defined in the Variable Manager will be automatically generated at runtime.")]
         public string v_OutputUserVariableName { get; set; }
 
+        [XmlIgnore]
```
[the remark this answers is not in the document]
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.Outlook/GetOutlookEmailsCommand.cs]: done

**Review by openbots-ch — DISMISSED** [review]

**Review by openbots-ch — APPROVED** [review]

File: OpenBots.Commands/OpenBots.Commands.Word/WordCreateApplicationCommand.cs [hunk 4]
```diff
@@ -134,5 +144,23 @@ public override string GetDisplayValue()
         {
             return base.GetDisplayValue() + $" [{v_NewOpenDocument} - Visible '{v_Visible}' - Close Instances '{v_CloseAllInstances}' - New Instance Name '{v_InstanceName}']";
         }
+
+        private void OpenFileComboBox_SelectedValueChanged(object sender, EventArgs e)
+        {
+            if (((ComboBox)RenderedControls[3]).Text == "Open Workbook")
```
Comment (openbots-ab) [inline OpenBots.Commands/OpenBots.Commands.Word/WordCreateApplicationCommand.cs:150]:
```suggestion
            if (((ComboBox)RenderedControls[3]).Text == "Open Document")
```

File: OpenBots.Commands/OpenBots.Commands.TextFile/WriteTextFileCommand.cs [hunk 5]
```diff
@@ -9,48 +9,47 @@
...
-using System.Xml.Serialization;
 
 namespace OpenBots.Commands.TextFile
 {
     [Serializable]
     [Group("Text File Commands")]
-    [Description("This command writes specified data to a text file.")]
-    public class WriteTextFileCommand : ScriptCommand
+    [Description("This command writes specified data to an existing or newly created text file.")]
+    public class WriteCreateTextFileCommand : ScriptCommand
```
Comment (openbots-ab) [inline OpenBots.Commands/OpenBots.Commands.TextFile/WriteTextFileCommand.cs:18]: Please update the File name to match with the Class name.

**Review by openbots-ab — CHANGES_REQUESTED** [review]
A couple of changes need to be made.

File: OpenBots.Commands/OpenBots.Commands.TextFile/WriteTextFileCommand.cs (hunk 5 above)
Comment (openbots-ff) [reply OpenBots.Commands/OpenBots.Commands.TextFile/WriteTextFileCommand.cs:18]: done

**Review by openbots-ab — APPROVED** [review]

**Review by openbots-ch — APPROVED** [review]

**Pull request merged**